### ISSUE TITLE
Using JOIN with ON/USING keyword fails to determine source tables when followed by a parenthesis

### ISSUE THREAD

**Issue by snowmanwangyal** [reata/sqllineage#233]
I have the following queries:

```
create table target_table as (
    select *
    from source_table_a
    join source_table_b
    on (source_table_a.some_id = source_table_b.some_id))
)
```
```
create table target_table as (
    select *
    from source_table_a
    join source_table_b
    using (some_id)
)
```
This causes the table lineage to not extract `source_table_a` and `source_table_b` as sources. However, when the parenthesis are removed around `source_table_a.some_id = source_table_b.some_id)` and `some_id` it works properly.

Any help or insight would be greatly appreciated! 😀

**Comment by snowmanwangyal** [comment]
I didn't spend too much time investigating, but I noticed from the `sqlparse` side when it is parsing the queries, it is treating everything inside as a Parenthesis whereas in the queries without the parenthesis it is treated as an Identifier.

I wonder where the logic lies where not having an identifier or having a parenthesis breaks it.

**Comment by reata** [comment]
> I didn't spend too much time investigating, but I noticed from the `sqlparse` side when it is parsing the queries, it is treating everything inside as a Parenthesis whereas in the queries without the parenthesis it is treated as an Identifier.
> 
> I wonder where the logic lies where not having an identifier or having a parenthesis breaks it.

Generally it's OK for sqlparse to generate AST like that because it's a non-validating parser. We already have a bunch of logic in sqllineage to handle various edge cases like this. 

This issue is potentially related with #213

**Comment by reata** [comment]
Confirmed this issue is introduced by PR #223 when I tried to strip the nested parenthesis around subquery. It's released with v1.3.4. Version 1.3.3 is fine. 


More details:
```sql
SELECT * FROM ((SELECT * FROM foo))
```

We have a buggy implementation by not asserting the parenthesis to be stripped at the beginning of the token. So in this case:
```sql
create table target_table as (
    select *
    from source_table_a
    join source_table_b
    on (source_table_a.some_id = source_table_b.some_id))
)
```

when inner parenthesis present after JOIN (or USING), it's identified as parenthesis around subquery, which is clearly wrong.
